homeassistant/components/ecoforest/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""The Ecoforest integration."""
+from __future__ import annotations
+
+import logging
+
+import async_timeout
+import httpx
+from pyecoforest.api import EcoforestApi
+from pyecoforest.exceptions import EcoforestAuthenticationRequired
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ecoforest from a config entry."""
+
+    host = entry.data[CONF_HOST]
+    auth = httpx.BasicAuth(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+    api = EcoforestApi(host, auth)
+
+    try:
+        async with async_timeout.timeout(10):
+            device = await api.get()
+            _LOGGER.debug("Ecoforest: %s", device)
+    except EcoforestAuthenticationRequired:
+        _LOGGER.error("Authentication on device %s failed", host)
+        return False
+    except Exception as err:  # pylint: disable=broad-except
+        _LOGGER.error("Error communicating with device %s", host)
+        raise ConfigEntryNotReady from err
```

Comment:
I just asked someone for a quick double check, and we only allow broad excepts in the config flow. So I think we should catch the appropriate exceptions here.

Reply:
I will wait for feedback then...

Reply:
This was the feedback haha

Reply:
So instead of the broad except we should catch the exceptions that can be raised here

Reply:
The library only raises the `EcoforestAuthenticationRequired`. other than that it could appear httpx exceptions. Should we catch all httpx exceptions?

Reply:
Hmm, good question. I have no real answer to this.

I do have an alternative, and that would be catching the httpx exceptions in the lib and raising your own exceptions. I did this with aiohttp exceptions in opensky: https://github.com/joostlek/python-opensky/blob/894c0f74b0f19fbbc7b1c90f7ed252b49b64ddc5/src/python_opensky/opensky.py#L128

I think this way I can be confident that we can make it work. (and if you are adding stuff to the lib, you can directly add the timeout)

Reply:
ok, so can I do it later in another PR or does it need to be now?

Reply:
I think this is a requirement in that sense, so I would say now. And this is literally the only thing left for this PR imo